src/MoBi.UI/Views/EditIndividualAndExpressionConfigurationsView.cs

```diff
@@ -59,16 +72,27 @@ public override void InitializeBinding()
             .WithImages(individual => _imageListRetriever.ImageIndex(individual.Icon))
             .WithValues(x => x.AllIndividuals);
 
-         btnAdd.Click += (o, e) => OnEvent(addSelectedExpression);
-         btnRemove.Click += (o, e) => OnEvent(removeSelectedExpression);
+         btnAdd.Click += (o, e) => OnEvent(addSelectedExpressions);
+         btnRemove.Click += (o, e) => OnEvent(removeSelectedExpressions);
 
-         projectExpressionsTree.TreeView.SelectedNodeChanged += selectedNode => OnEvent(() => _presenter.ProjectExpressionSelectionChanged(selectedNode));
-         simulationExpressionsTree.SelectedNodeChanged += selectedNode => OnEvent(() => _presenter.SimulationExpressionSelectionChanged(selectedNode));
+         projectExpressionsTree.TreeView.SelectionChanged += (sender, args) =>
+            projectExpressionsTree.TreeView.SelectionChanged += (o, e) => OnEvent(projectTreeSelectionChanged);
+         simulationExpressionsTree.SelectionChanged += (o, e) => OnEvent(simulationTreeSelectionChanged);
 
          simulationExpressionsTree.MouseDown += (o, e) => OnEvent(TreeMouseDown, e);
          simulationExpressionsTree.MouseMove += (o, e) => OnEvent(TreeMouseMove, e);
       }
 
+      private void simulationTreeSelectionChanged()
+      {
+         _presenter.SimulationExpressionSelectionChanged(treeViewSelectionToTreeNodeList(simulationExpressionsTree.Selection));
+      }
+
+      private void projectTreeSelectionChanged()
+      {
+         _presenter.ProjectExpressionSelectionChanged(treeViewSelectionToTreeNodeList(projectExpressionsTree.TreeView.Selection));
```

Comment:
why are those two object (simulationExpressionsTree and proectExpressionsTree) not symmetrical? One is a tree, one has a tree. Suspicious

Reply:
One is a filtered tree, the other is not filtered. I think the filtered one is created by us as a treeview and a textbox or something.